### PR TITLE
update badge to reflect visitor, update library locations for visitors

### LINKED ISSUE #2162: Restrict pickup locations for name/email patrons
Users should NOT be able to select Law or Business. They should definitely be able to select Green, East Asia, Art, and Lane. (Sarah is confirming the rest of the SUL libraries.)

### PR DESCRIPTION
**Pull request by dnoneill**
closes #2162

### CODE REVIEW

File: app/views/patron_requests/_user_header.html.erb [hunk 1]
```diff
@@ -9,7 +9,7 @@
       <div class="d-flex align-items-center">
         <div class="h4">Welcome <%= current_user.patron.display_name %>!</div>
         <span class="badge ms-2 mb-2 rounded-4 bg-jungle-green">
-          <%= current_user.patron.patron_group['desc'] %>
+          <%= current_user.patron&.id == 'visitor' ? 'Visitor' : current_user.patron.patron_group['desc'] %>
```
Comment (cbeer) [inline app/views/patron_requests/_user_header.html.erb:12]: It looks like there's a method on the patron model for this?

```suggestion
          <%= current_user.patron.visitor_patron? ? 'Visitor' : current_user.patron.patron_group['desc'] %>
```

File: config/settings.yml [hunk 2]
```diff
@@ -663,4 +663,8 @@ paging_schedule:
 hold_instead_of_recall:
   - 'In process'
   - 'On order'
-  - 'In transit'
\ No newline at end of file
+  - 'In transit'
+
+unallowed_visitor_pickups:
+  - 'BUS-IDESK'
+  - 'LAW'
```
Comment (cbeer) [inline config/settings.yml:670]: I think we probably want to allow-list the ok pickup locations? I think there's only a couple pickup locations that can also handle visitor registration, right?

File: app/models/patron_request.rb [hunk 3]
```diff
@@ -45,6 +45,8 @@ def pickup_service_point
 
   # FOLIO
   def pickup_destinations
+    return Settings.allowed_visitor_pickups if patron.visitor_patron?
```
Comment (cbeer) [inline app/models/patron_request.rb:48]: This still needs to respect the location_restricted_service_point_codes, too, right? (e.g. PAGE-EA can only be picked up in EAST-ASIA)

Comment (dnoneill) [reply app/models/patron_request.rb:48]: Yup. Thanks.

**Pull request merged**